BinaryEncoding.md

```diff
@@ -786,9 +783,140 @@ for [future :unicorn:][future multiple tables] use and must be 0 in the MVP.
+| `i64.atomic_xor32_s` | `0xe61a` | `4` `memory immediate` | atomic xor with memory |
+| `i64.atomic_xor32_u` | `0xe61b` | `4` `memory immediate` | atomic xor with memory |
+| `i32.atomic_xchg` | `0xe610` | `5` `memory immediate` | atomic exchange with memory |
+| `i64.atomic_xchg` | `0xe611` | `5` `memory immediate` | atomic exchange with memory |
+| `i32.atomic_xchg8_s` | `0xe612` | `5` `memory immediate` | atomic exchange with memory |
+| `i32.atomic_xchg8_u` | `0xe613` | `5` `memory immediate` | atomic exchange with memory |
+| `i32.atomic_xchg16_s` | `0xe614` | `5` `memory immediate` | atomic exchange with memory |
+| `i32.atomic_xchg16_u` | `0xe615` | `5` `memory immediate` | atomic exchange with memory |
+| `i64.atomic_xchg8_s` | `0xe616` | `5` `memory immediate` | atomic exchange with memory |
+| `i64.atomic_xchg8_u` | `0xe617` | `5` `memory immediate` | atomic exchange with memory |
+| `i64.atomic_xchg16_s` | `0xe618` | `5` `memory immediate` | atomic exchange with memory |
+| `i64.atomic_xchg16_u` | `0xe619` | `5` `memory immediate` | atomic exchange with memory |
+| `i64.atomic_xchg32_s` | `0xe61a` | `5` `memory immediate` | atomic exchange with memory |
+| `i64.atomic_xchg32_u` | `0xe61b` | `5` `memory immediate` | atomic exchange with memory |
+
+## Atomic compare exchange operators ([described here](Semantics.md#atomic-memory-accesses))
+
+| Name | Opcode | Immediate | Description |
+| ---- | ---- | ---- | ---- |
+| `i32.atomic_cmpxchg` | `0xe61c` | `memory immediate` | atomic compare exchange with memory |
+| `i64.atomic_cmpxchg` | `0xe61d` | `memory immediate` | atomic compare exchange with memory |
+| `i32.atomic_cmpxchg8_s` | `0xe61e` | `memory immediate` | atomic compare exchange with memory |
+| `i32.atomic_cmpxchg8_u` | `0xe61f` | `memory immediate` | atomic compare exchange with memory |
+| `i32.atomic_cmpxchg16_s` | `0xe620` | `memory immediate` | atomic compare exchange with memory |
+| `i32.atomic_cmpxchg16_u` | `0xe621` | `memory immediate` | atomic compare exchange with memory |
+| `i64.atomic_cmpxchg8_s` | `0xe622` | `memory immediate` | atomic compare exchange with memory |
+| `i64.atomic_cmpxchg8_u` | `0xe623` | `memory immediate` | atomic compare exchange with memory |
+| `i64.atomic_cmpxchg16_s` | `0xe624` | `memory immediate` | atomic compare exchange with memory |
+| `i64.atomic_cmpxchg16_u` | `0xe625` | `memory immediate` | atomic compare exchange with memory |
+| `i64.atomic_cmpxchg32_s` | `0xe626` | `memory immediate` | atomic compare exchange with memory |
+| `i64.atomic_cmpxchg32_u` | `0xe627` | `memory immediate` | atomic compare exchange with memory |
+
+## Atomic load/store operators ([described here](Semantics.md#atomic-memory-accesses))
+
+| Name | Opcode | Immediate | Description |
+| ---- | ---- | ---- | ---- |
+| `i32.atomic_load` | `0xe628` | `memory immediate` | atomic load from memory |
+| `i64.atomic_load` | `0xe629` | `memory immediate` | atomic load from memory |
```

Comment:
Yes, and I believe that's what was discussed before. But I had a thought it might be nice to have all atomic/threading operators use the same prefix, which led me down this route. One additional nicety is that the opcodes for atomic load/store operators match the non-atomic ones, e.g. `i32.atomic_load` is `0xe628` and `i32.load` is `0x28`, etc.

Since the opcode specifies that it is atomic, this also means that we don't need a special "non-atomic" value for the memory immediate for RMW ops too.

But I'm not married to the idea, just thought it was kinda neat. :-)